airflow/providers/google/cloud/operators/cloud_storage_transfer_service.py

```diff
@@ -546,28 +547,29 @@ def __init__(
     ) -> None:
         # To preserve backward compatibility
         # TODO: remove one day
+        _filter = kwargs.pop("filter", None)
         if request_filter is None:
-            if "filter" in kwargs:
-                request_filter = kwargs["filter"]
-                AirflowProviderDeprecationWarning(
-                    "Use 'request_filter' instead 'filter' to pass the argument."
-                )
-            else:
-                TypeError("__init__() missing 1 required positional argument: 'request_filter'")
-
+            if not _filter:
+                raise ValueError("'request_filter' should be set")
+            request_filter = _filter
+            warnings.warn(
+                "Use 'request_filter' instead 'filter' to pass the argument.",
+                AirflowProviderDeprecationWarning,
+                stacklevel=2,
+            )
         super().__init__(**kwargs)
         self.filter = request_filter
         self.project_id = project_id
         self.gcp_conn_id = gcp_conn_id
         self.api_version = api_version
         self.google_impersonation_chain = google_impersonation_chain
-        self._validate_inputs()
 
     def _validate_inputs(self) -> None:
         if not self.filter:
             raise AirflowException("The required parameter 'filter' is empty or None")
 
     def execute(self, context: Context) -> list[dict]:
+        self._validate_inputs()
```

Comment:
Small clarification. Why we moved _validate_inputs from init to execute method?

Reply:
Templated field could rendered into the empty value, and we know it only after render this value